main/main.cpp

```diff
@@ -3967,6 +3967,11 @@ bool Main::iteration() {
 		PhysicsServer3D::get_singleton()->flush_queries();
 #endif // _3D_DISABLED
 
+		// Prepare the fixed timestep interpolated nodes BEFORE they are updated
+		// by the physics server, otherwise the current and previous transforms
+		// may be the same, and no interpolation takes place.
+		OS::get_singleton()->get_main_loop()->iteration_prepare();
```

Comment:
If the goal is to do this before physics, why isn't it done before all physics (including 3D physics?)

Reply:
@aaronfranke You are also absolutely right, I didn't spot this in the PR, the prepare should probably be before the 3D here (although it shouldn't make any difference until the 3D is done, so we can fix in 3D PR).

To address the confusion a little - "physics interpolation" (somewhat confusingly) is nothing to do with physics engine. The more correct name is fixed timestep interpolation.

The only relation is to the physics ticks, rather than the physics itself. It can work with or without physics engine. Juan preferred the name "physics interpolation", because even though incorrect, he thought it would be easier for users to understand (and admittedly in many cases their eyes would glaze over with "fixed timestep interpolation" :grin: ).

This is something that we as developers will be expected to learn the difference, but it will inevitably cause confusion in some way (if we e.g. called it "fixed timestep interpolation" in the c++, and "physics interpolation" for user facing code, this could be confusing, as many c++ funcs are bound etc, so it is called "physics interpolation" throughout).

As it is not related to the physics engines, it should not be disabled when disabling physics engines (as `_physics_process()` still occurs).

Just to try and explain what it is doing .. the `VisualServer` items have to be prepared before each iteration (tick) before any changes have been made to them (server side at least).

If you follow the code it calls `update_interpolation_tick()`.

This does a bunch of things:
* Detect any objects that are no longer being moved, and ensure their previous transform = current transform so they cease vibrating
* Store the current transform (from the last tick) to the previous transform, so they can be interpolated after the tick